Common/Securities/SecurityPortfolioManager.cs

```diff
@@ -749,6 +775,7 @@ public void ScanForCashSettlement(DateTime timeUtc)
 
                         // update settled cashbook
                         CashBook[item.Currency].AddAmount(item.Amount);
+                        InvalidateTotalPortfolioValue();
```

Comment:
Ideally can move internally so cashbook can trigger invalidations himself.

Reply:
I think ideally no external component should know that this invalidation is happening, nor should they care. One way of handling this is to have components broadcast when they have updated. So when someone adds money to the cashbook, the cashbook raises his `BalancesChanged` event (or whatever it may be). The portfolio, upon construction, can bind to that event, `cashBook.BalancesChanged += InvalidateTotalPortfolioValue();` so each time the cash book changes, he can perform the invalidation -- a similar pattern can be applied to other dynamic data.

This could also lend itself towards even _more_ efficient algorithms (here and elsewhere), by permitting more incremental algorithms which could show huge benefits for algorithms with many symbols

Reply:
Good idea 👍  addressed in separate PR https://github.com/QuantConnect/Lean/pull/3282, for now, specifically for `Cash` and `CashBook` updates